### ISSUE TITLE
Is it a feature? A bug, maybe?

### ISSUE THREAD

**Issue by ArtixKrieg** [diantahoc/chan-archiver#6]
Deleting a thread via the Web interface does not delete the images and other files associated with it from the computer (although the thread is no longer accessible, despite the files remaining on my hard drive). Was this functionality never implemented, or is it a bug?

Also, I may be doing something wrong, but the Command Line switches (such as '--savedir') do not work with the Interactive Console. I would like to change the save directory but I can't find a way to do that. Any help would be appreciated.

Windows 7, Chrome, Firefox

**Comment by diantahoc** [comment]
Deleting a thread should delete it's files (.json files) and the images IF there is no other thread using the same images. I will investigate this issue though, I might have introduced a bug.

The interactive console cannot change the save directory. You should start ChanArchiver with `--savedir <dir>` instead. The `<dir>` parameter is expected to be after `--savedir`, otherwise it won't work. If you are doing this and it's still not working, then please explain in details what are you doing.

**Comment by ArtixKrieg** [comment]
I tested this issue again using thread no. 291485538 on /v/ and no. 47519808 on /g/ for my tests. 

Once a thread was archived, I deleted it via the Web Interface using Chrome, then downloaded the images with 'getfilelist' in Firefox using the DownThemAll add-on. All but one image was downloaded successfully both times. The images which aren't being downloaded are not listed in the getfilelist and I can't locate the filename in my files.

How do I start ChanArchiver with savedir? Am I supposed to append these switches to the .exe on startup?

Thanks.

**Comment by diantahoc** [comment]
I still haven't investigated the issue due to lack of time, but to answer you question, yes you should pass these switches to the executable at startup.

**Comment by diantahoc** [comment]
Please download this new release and see if the problem persist.

https://github.com/diantahoc/chan-archiver/releases/tag/1.11

**Comment by ArtixKrieg** [comment]
Crashes on startup.
![chanarchiver](https://cloud.githubusercontent.com/assets/11968443/7338812/c16cb024-ec0c-11e4-9786-5e2db252b1d5.jpg)

**Comment by diantahoc** [comment]
Go to ChanArchiver save directory and delete the `aniwrap_cache` folder.

**Comment by ArtixKrieg** [comment]
Thanks! I discovered "file-index.json" was causing ChanArchiver to crash a lot, so I deleted that as well.

One more thing. I can't view any boards I've archived or the threads associated with them. Every time I try, I get this error:
![cannot access board](https://cloud.githubusercontent.com/assets/11968443/7341999/6256a982-ec61-11e4-8591-fae3a0a4ac75.jpg)

Deleting files doesn't seem to fix the problem. Any ideas?

**Comment by diantahoc** [comment]
I don't know why it's happening, but it looks like this problem is related to your environment, as it does not happen on my machine even with the latest version. What is causing this is that the `DateTime` serializer is using a parenthesis `)` character instead of the number of hours in the time zone.

Either way, I have changed the way that it is serialized, so it should work now.

Try the `Release-Apr29.zip` file. Thank you for your patience and feedback.

**Comment by ArtixKrieg** [comment]
Problem solved! Thank you for all your work!

**Comment by diantahoc** [comment]
Yes, it's not working for me either. I am currently trying to add a web interface for it, so it's easier to use. I will notify you when it's done.

**Comment by ArtixKrieg** [comment]
Great idea. While you're at it, what do you think about adding a button to export a zip file of images from a specified thread? Or better yet, the entire thread? I don't know how hard this would be to implement, or what practicality it has for the average person, but I would find the feature useful for myself. The button could be added to a thread next to [File List] and under the "Watch Jobs" page.

![example](https://cloud.githubusercontent.com/assets/11968443/7428844/7999c572-efac-11e4-92e4-1c01ceb0c90c.jpg)

**Comment by diantahoc** [comment]
Not a bad idea. I will try to do it.

**Comment by diantahoc** [comment]
I finally implemented it.

![breads](https://cloud.githubusercontent.com/assets/2142305/7785032/4612e1a2-0186-11e5-92a7-3cda7a831085.png)

**Comment by diantahoc** [comment]
About the export as zip button, I will also notify you when it's done.

**Comment by ArtixKrieg** [comment]
Great. Everything works on my end. Thanks for fixing that!

I noticed there was a note feature for thread filters. Could you add that to individual threads as well? Maybe by including a [Add Note] button next to [Show Info] and [Ban File], which would append a little text above the thread.

Something like this.
![add note 2](https://cloud.githubusercontent.com/assets/11968443/7808572/19b1ee72-0349-11e5-9f17-d074fac4fb8a.png)

Or maybe.
![add note 1](https://cloud.githubusercontent.com/assets/11968443/7808579/268ac83a-0349-11e5-87fc-00850d0b11d2.png)

**Comment by diantahoc** [comment]
Yes of course.

**Comment by ArtixKrieg** [comment]
I only just noticed, but 404'ed threads which have already been archived can't be updated from external 4chan archives. I've tried it with both an old and new install of ChanArchiver. Same issue.

To try it yourself, just add a thread and then stop it. Make a few posts and add some images, then try updating the thread using an external 4chan archive. If I'm not the only one with this problem, you'll notice the thread won't be updated.

**Comment by diantahoc** [comment]
Good observation. I will check into it when I have time.

**Comment by diantahoc** [comment]
I have added download as zip feature and thread notes.

Get the latest release from here:

https://github.com/diantahoc/chan-archiver/releases/tag/1.12

**Comment by ArtixKrieg** [comment]
Works fantastic. Thank you! Instead of deleting threads to save storage space, I can now download them and store the file elsewhere. My poor SSD lives another day.

I was experimenting with the Command Line Switches today when I noticed an issue occurs with --savedir. Everything seems to work as it should... All files are still downloading successfully, thumbnails are showing, image files appear in the getfile list and I can download them in full resolution, but they aren't displaying in the browser. Might this be because ChanArchiver is trying to fetch the images from the old directory instead of the new one?

And while trying to debug the issue, I also noticed --thumbonly and "Only save thumbnails" in Settings doesn't work in preventing full resolution images downloads. Which isn't important to me, but I thought I would mention it.

**Comment by diantahoc** [comment]
I am currently trying to organize ChanArchiver code. This might get rid of (or maybe add) a few unnoticed bugs. Thanks for your feedback, I really appreciate it.
